lib/bolt/pal.rb

```diff
@@ -52,12 +52,21 @@ def initialize(modulepath, hiera_config, resource_types, max_compiles = Etc.npro
       @apply_settings = apply_settings
       @max_compiles = max_compiles
       @resource_types = resource_types
+      @project = project
 
       @logger = Logging.logger[self]
       if modulepath && !modulepath.empty?
         @logger.info("Loading modules from #{@modulepath.join(File::PATH_SEPARATOR)}")
       end
 
+      # If the project name is the same as one of the builtin modules raise a warning
+      bolt_mods = Dir.children(BOLTLIB_PATH)
+      if @project && bolt_mods.include?(@project.name)
+        msg = "The project #{@project.name} will not load top-level content as it has a reserved name: #{bolt_mods}"
```

Comment:
This message comes out a little weird.

```
The project boltlib will not load top-level content as it has a reserved name: ["file", "out", "ctrl", "system", "boltlib", "prompt"]
```

Not quite enough context is presented for _why_ that list of names is there.

Reply:
Boltlib is especially weird because I can even say "builtin plan modules" since we alias boltlib functions and plans to nix the 'boltlib' prefix. Would just `"The project boltlib will not load top-level content as it conflicts with builtin Bolt modules: ["file", "out", "ctrl", "system", "boltlib", "prompt"]"` be better?

Reply:
I guess I'm not sure it helps much to include all the options. That would be like... the natural name for your project conflicts with a built-in name, but make sure you don't accidentally pick one of these _other_ (very unrelated) built-in names when you change it.

"Project-level content will not be loaded because the  project name 'boltlib' conflicts with a built-in module of the same name" perhaps?

Reply:
I'm also trying to think through cases where it makes sense to have a project whose name conflicts with a built-in module even though the project content won't be loaded, ie. why should this be a warning and not just an error. I guess it's mainly if you have a project without `name` set in project.yaml, but that happens to be in a directory that happens to be the same as a built-in module (unlikely as that is) and therefore the default name would conflict.

Other than that, the whole point of giving a project a name is to load its top-level content. You might not care about loading the top-level content if you're just developing the project to ship it as a module, but in that case it won't work as a module _either_ (since it will be shadowed by the built-in module of the same name).

Reply:
This makes sense, I moved the validation to be in the project validation.